CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/file/DownloadImageUtilitiesNGTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2010-2021 Australian Signals Directorate
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.file;
+
+import au.gov.asd.tac.constellation.utilities.https.HttpsConnection;
+import au.gov.asd.tac.constellation.utilities.https.HttpsUtilities;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.TimeoutException;
+import javafx.scene.image.Image;
+import javax.net.ssl.HttpsURLConnection;
+import org.mockito.MockedStatic;
+import org.testng.annotations.Test;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import org.testfx.api.FxToolkit;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * @author groombridge34a
+ */
+public class DownloadImageUtilitiesNGTest {
+    
+    private static final String TEST_IMAGE = "pixel.png";
+    
+    /**
+     * Can get an image from the specified URL.
+     * 
+     * @throws IOException if mocking the HTTPS connection classes fails.
+     */
+    @Test
+    public void testGetImage() throws IOException {
+        try (
+                final MockedStatic<HttpsConnection> connMockedStatic = mockStatic(HttpsConnection.class);
+                final MockedStatic<HttpsUtilities> utilsMockedStatic = mockStatic(HttpsUtilities.class);) 
+        {
+            // set up mocks
+            final HttpsURLConnection urlConn = mock(HttpsURLConnection.class);
+            when(urlConn.getResponseCode()).thenReturn(HttpsURLConnection.HTTP_OK);
+            
+            final HttpsConnection conn = mock(HttpsConnection.class);
+            when(conn.acceptPng()).thenReturn(conn);
+            when(conn.withReadTimeout(anyInt())).thenReturn(conn);
+            when(conn.get()).thenReturn(urlConn);
+            
+            connMockedStatic.when(() -> HttpsConnection.withUrl(anyString())).thenReturn(conn);
```

Comment:
I assume the URL is known? Or can be set up via configuration in the test setup so that this can be verified here.

Reply:
My bad, I got so into wiring up the mocks I didn't pass the method param through 😊